library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
Let me double check if I understood your point. You are suggesting the type should be:
```lean
@[reducible] meta def parse {α : Type} (p : lean_parser pexpr) : Type := α
```
where the meaning is: `p` is a parser action that produces a `pexpr` which should have type `α` when elaborated. If it doesn't, a nasty error message is produced. That is, it is up to the parser extension/action developer to enforce this property.